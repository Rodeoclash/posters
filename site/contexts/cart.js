import { createContext, useContext } from "react";

const CartContext = createContext({
  add: () => {},
  contains: () => {},
  hide: () => {},
  remove: () => {},
  show: () => {},
  showing: false,
  size: 0,
  variants: [],
});

export const useCartContext = () => useContext(CartContext);

export default CartContext;
