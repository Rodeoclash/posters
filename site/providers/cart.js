import { useState } from "react";
import CartContext from "../contexts/cart.js";

const LoadingProvider = ({ children }) => {
  const [state, setState] = useState({
    products: [],
  });

  const addProduct = (product) => {
    setState({
      ...state,
      products: state.products.concat([product]),
    });
  };

  const value = {
    addProduct,
    products: state.products,
    size: state.products.length,
  };

  return <CartContext.Provider value={value}>{children}</CartContext.Provider>;
};

export default LoadingProvider;
