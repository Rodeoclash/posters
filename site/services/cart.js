export class ProductNotFound extends Error {}
export class ProductExists extends Error {}

const productMatch = (product1, product2) => {
  return product1.id === product2.id;
};

const productNoMatch = (product1, product2) => {
  return !productMatch(product1, product2);
};

export const show = (cart) => {
  return {
    ...cart,
    showing: true,
  };
};

export const hide = (cart) => {
  return {
    ...cart,
    showing: false,
  };
};

export const contains = (cart, product) => {
  return cart.products.some(productMatch.bind(this, product));
};

export const add = (cart, product) => {
  if (contains(cart, product) === true) {
    throw new ProductExists();
  }

  return {
    ...show(cart),
    products: cart.products.concat([product]),
  };
};

export const remove = (cart, product) => {
  if (contains(cart, product) === false) {
    throw new ProductNotFound();
  }

  return {
    ...show(cart),
    products: cart.products.filter(productNoMatch.bind(this, product)),
  };
};
