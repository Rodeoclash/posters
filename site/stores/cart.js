import { createContext } from "react";
import { useObservable } from "mobx-react-lite";

const CartContext = createContext();

function Cart({ children }) {
  const store = useObservable({ counter: 0 });
  return <CartContext.Provider value={store}>{children}</CartContext.Provider>;
}

export { Cart, CartContext };
