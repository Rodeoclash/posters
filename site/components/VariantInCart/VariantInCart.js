import { useCartContext } from "../../contexts/cart";

const VariantInCart = ({ variant }) => {
  return null;

  const { contains } = useCartContext();

  if (contains(variant) === false) {
    return null;
  }

  return <em>This item is in your cart</em>;
};

export default VariantInCart;
