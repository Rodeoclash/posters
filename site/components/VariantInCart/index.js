import VariantInCart from "./VariantInCart";
export default VariantInCart;
