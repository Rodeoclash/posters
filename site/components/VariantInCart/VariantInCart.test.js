import React from "react";
import renderer from "react-test-renderer";

import * as CartContext from "../../contexts/cart";
import VariantInCart from "./VariantInCart";

describe("snapshots", () => {
  it("with variant in the cart", () => {
    const variant = {};

    const contextValues = {
      contains() {
        return true;
      },
    };

    jest
      .spyOn(CartContext, "useCartContext")
      .mockImplementation(() => contextValues);

    const tree = renderer.create(<VariantInCart variant={variant} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("with variant not in the cart", () => {
    const variant = {};

    const contextValues = {
      contains() {
        return false;
      },
    };

    jest
      .spyOn(CartContext, "useCartContext")
      .mockImplementation(() => contextValues);

    const tree = renderer.create(<VariantInCart variant={variant} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
