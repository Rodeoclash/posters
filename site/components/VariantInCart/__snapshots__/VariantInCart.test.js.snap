// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots with variant in the cart 1`] = `null`;

exports[`snapshots with variant not in the cart 1`] = `null`;
